### PR TITLE
Replace lifecycled without SNS/SQS

### PR DESCRIPTION
**Pull request by sj26**
We've had some trouble with SQS lifecycle handling, so we can try something a little less sophisticated and _POLL FOR GREAT JUSTICE_. Maybe. Haha.

It definitely works:

screenshot 2016-09-01 18 09 23

### CODE REVIEW

File: templates/buildkite-elastic.yml [hunk 1]
```diff
@@ -442,22 +466,28 @@ Resources:
             06-run-bootstrap-script:
               test: test -n "$(BootstrapScriptUrl)"
               command: |
-                #!/bin/bash -euo pipefail
```
Comment (sj26) [inline templates/buildkite-elastic.yml]: (This option caused an error for me)

**Comment by toolmantim** [comment]
Fixes #130

**Comment by toolmantim** [comment]
Stack JSON URL: `https://s3.amazonaws.com/buildkite-aws-stack/lifecycled-the-second/aws-stack.json`

**Comment by lox** [comment]
:(

**Comment by toolmantim** [comment]
Sorry @lox! Turns out you can do a `CONTINUE` without a lifecycle action token, and just the instance id. The docs/guide/examples don't make that obvious at all.

Which means you can just poll your own state to see the state change.

File: templates/buildkite-elastic.yml [hunk 2]
```diff
@@ -428,7 +428,7 @@ Resources:
                                        --query AutoScalingInstances[].LifecycleState --output text)"
                   logger -t lifecycled "Lifecycle state for \$INSTANCE_ID: \$LIFECYCLE_STATE"
                   if [[ "\$LIFECYCLE_STATE" == "Terminating:Wait" ]]; then
-                    logger -t lifecycled "Time to shut it all down!"
```
Comment (sj26) [inline templates/buildkite-elastic.yml]: Oh sorry, this would have broken. Woops!

Comment (sj26) [inline templates/buildkite-elastic.yml]: Oh no, it did work in the example. Perhaps it's too enthusiastic, though.

**Comment by toolmantim** [comment]
Rolling updates work AOK, scaling up and down works AOK. I think this is good to 🚀

**Comment by toolmantim** [comment]
This seems to be working well 👍

**Pull request merged**